web/src/copy_and_paste.ts

```diff
@@ -402,6 +402,34 @@ export function paste_handler_converter(paste_html: string): string {
         .parseFromString(paste_html, "text/html")
         .querySelector("body");
     assert(copied_html_fragment !== null);
+
+    const para_elements = copied_html_fragment.querySelectorAll("p");
+
+    for (const element of para_elements) {
+        const children = [...element.childNodes];
+
+        for (let i = 0; i < children.length - 1; i += 1) {
+            const current_node = children[i];
+            const next_node = children[i + 1];
+
+            if (!current_node || !next_node) {
+                continue;
+            }
+
+            if (
+                next_node.nodeType === 3 // Check if the next node is a text node
```

Comment:
```suggestion
                next_node.nodeType === Node.TEXT_NODE
```

Reply:
The tests are always lies because they mock everything. Fix the tests to mock this with the `Node` from JSDOM (`global.Node = window.Node`).